src/interfaces/chain.cpp

```diff
@@ -0,0 +1,438 @@
+#include <interfaces/chain.h>
+
+#include <amount.h>
+#include <chain.h>
+#include <chainparams.h>
+#include <interfaces/handler.h>
+#include <net.h>
+#include <policy/feerate.h>
+#include <policy/policy.h>
+#include <policy/rbf.h>
+#include <primitives/block.h>
+#include <primitives/transaction.h>
+#include <protocol.h>
+#include <rpc/mining.h>
+#include <rpc/protocol.h>
+#include <rpc/server.h>
+#include <sync.h>
+#include <timedata.h>
+#include <txmempool.h>
+#include <ui_interface.h>
+#include <uint256.h>
+#include <univalue.h>
+#include <util.h>
+#include <validation.h>
+#include <validationinterface.h>
+
+#if defined(HAVE_CONFIG_H)
+#include <config/bitcoin-config.h>
+#endif
+#ifdef ENABLE_WALLET
+#include <wallet/fees.h>
+#include <wallet/wallet.h>
+#define CHECK_WALLET(x) x
+#else
+#define CHECK_WALLET(x) throw std::logic_error("Wallet function called in non-wallet build.")
+#endif
+
+#include <algorithm>
+#include <future>
+#include <map>
+#include <memory>
```

Comment:
```
Include(s) from src/interfaces/chain.h duplicated in src/interfaces/chain.cpp:
#include <memory>
Include(s) from src/interfaces/wallet.h duplicated in src/interfaces/wallet.cpp:
#include <memory>
#include <string>
#include <utility>
#include <vector>
Include(s) from src/wallet/init.h duplicated in src/wallet/init.cpp:
#include <walletinitinterface.h>
^---- failure generated from contrib/devtools/lint-includes.sh

Reply:
> failure generated from contrib/devtools/lint-includes.sh

I'll try to submit a pr to simplify the developer guideline and this lint check. This check is incompatible with the [iwyu tool](https://github.com/include-what-you-use/include-what-you-use/blob/master/docs/WhyIWYU.md), and I think "include what you use" is a better and simpler policy than "include what you use most of the time but not if the same include line appears in a different file"

Reply:
@ryanofsky Agree with you here. I merged that pull request because it had some acks and no nack, afaics. I am happy to ACK a pull that adjusts the developer guidelines and removes the lint check.